lib/employee.ts

```diff
@@ -0,0 +1,55 @@
+import { AxiosRequestConfig } from "axios";
+
+export interface EmployeeResponse {
+    locationId: string;
+    posEmployeeId: string;
+    firstName: string;
+    lastName: string;
+    createdAt: string;
+    updatedAt: string;
+}
+
+export default class Employee {
+    readonly requestMaker: (data: AxiosRequestConfig) => Promise<any>;
+
+    constructor(requestMaker: (data: AxiosRequestConfig) => Promise<any>) {
+        this.requestMaker = requestMaker;
+    }
+
+    async get(
```

Comment:
Yeah, that's a good point. Although having all three does not necessarily mean they are redundant - it's merely exposing a range of intuitive methods one would expect from the SDK. And making it private doesn't add much value in terms of abstracting away details.

Reply:
It's certainly a good point. I agree with Muhammad in that it doesn't necessarily make the other functions redundant, it just makes it less obvious as to which function should be called.

Making it private does make it cleaner, but does it remove some flexibility that is a nice to have? I'm not sure of the answer, to be honest. I'll be guided by the 3 of you. I'm happy to make the change if we think this improve things - my only concern is that if we do remove it (and across other classes), it introduces a breaking change, so we'd need to move to `5.0.0` as other partners might be using them. :thinking:

Reply:
Ok, I really think `getAll` and `getOne` are redundant, mainly because the get method public accessible and has this flexibility by itself, however it's not a big deal for me.

Having said that, if you guys think that create a major version could be an overhead, then Im happy to keep as it is. But the only think I see as a **must** is JSDoc needed for all public methods for an SDK, so please let's add it to this get method in here.

Reply:
@johnatdoshii I don't think it should be a very massive undertaking. I'll have a look and create a ticket